crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -546,6 +546,88 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.all_transactions.txs_iter(sender).map(|(_, tx)| Arc::clone(&tx.transaction)).collect()
     }
 
+    /// Updates only the pending fees without triggering subpool updates.
+    /// Returns the previous base fee and blob fee values.
+    #[allow(clippy::missing_const_for_fn)]
+    fn update_pending_fees_only(
+        &mut self,
+        new_base_fee: u64,
+        new_blob_fee: Option<u128>,
+    ) -> (u64, u128) {
+        let prev_base_fee = self.all_transactions.pending_fees.base_fee;
+        let prev_blob_fee = self.all_transactions.pending_fees.blob_fee;
+
+        self.all_transactions.pending_fees.base_fee = new_base_fee;
+        if let Some(blob_fee) = new_blob_fee {
+            self.all_transactions.pending_fees.blob_fee = blob_fee;
+        }
+
+        (prev_base_fee, prev_blob_fee)
+    }
+
+    /// Applies fee-based promotion updates.
+    /// Directly modifies the provided outcome by pushing promoted transactions.
+    fn apply_fee_updates(
+        &mut self,
+        prev_base_fee: u64,
+        prev_blob_fee: u128,
+        outcome: &mut UpdateOutcome<T::Transaction>,
+    ) {
```

Comment:
this looks correct, but we still have

these functions

https://github.com/paradigmxyz/reth/blob/c45817c1f21b13368f75ffb1058dd1457d380f9f/crates/transaction-pool/src/pool/txpool.rs#L271-L271

https://github.com/paradigmxyz/reth/blob/c45817c1f21b13368f75ffb1058dd1457d380f9f/crates/transaction-pool/src/pool/txpool.rs#L224-L224

could we reuse them and give them `&mut UpdateOutcome`